bundle-workflow/src/manifests/build_manifest.py

```diff
@@ -54,6 +57,18 @@ def to_dict(self):
             ),
         }
 
+    @staticmethod
+    def get_build_manifest_relative_location(build_id, opensearch_version, architecture):
+        return f"builds/{opensearch_version}/{build_id}/{architecture}/manifest.yml"
+
+    @staticmethod
+    def from_s3(bucket_name, build_id, opensearch_version, architecture):
+        local_path = str(os.getcwd())
```

Comment:
This is going to leave files all around (you can't rely on the caller to be in a temporary directory). Download into a temporary file/path and automatically get rid of the file after loading it.

Reply:
the entire test logic executes in a tmp directory, which is automatically cleaned up after the test success/failure. These files as well get downloaded in the same tmp directory and get automatically cleaned up even for local runs. So it shouldn't leave files all over the place. However, I do see value in adding a cleanup inside this method, to make it complete in itself.

Reply:
You're making an assumption about something happening externally, which can change without this code ever noticing.

Reply:
>However, I do see value in adding a cleanup inside this method, to make it complete in itself.

Agree. i fixed it w/ cleanup logic in latest commit.